version: "3.8"

services:
  app_back:
    build:
      context: ./backend/
      dockerfile: Dockerfile
    image: lab4/back_end
    container_name: back_end
    env_file: .env
    command: uvicorn main:app --reload --workers 1 --host 0.0.0.0 --port 3000
    volumes:
      - ./backend/app:/code/app
    ports:
      - "8000:3000"
    depends_on:
      - database_lab4
    environment:
      - DATABASE_URL=postgresql://hello_fastapi:hello_fastapi@database_lab4/db_lab4
    networks:
      - mynetwork

  database_lab4:
    image: postgres:12.1-alpine
    container_name: database_lab4
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    environment:
      - POSTGRES_USER=hello_fastapi
      - POSTGRES_PASSWORD=hello_fastapi
      - POSTGRES_DB=db_lab4
    ports:
      - "5432:5432"
    networks:
      - mynetwork

  
  app_front:
    build:
      context: ./frontend/
      dockerfile: Dockerfile
      args:
        VITE_API_BACKEND: ${VITE_API_BACKEND}
    image: lab4/front_end
    container_name: front_end
    env_file:
      - .env
    environment:
      - VITE_API_BACKEND=${VITE_API_BACKEND}
    volumes:
      - ./frontend:/home/node/app:rw
    ports:
      - "3000:5173"
    restart: unless-stopped
    networks:
      - mynetwork
  
volumes:
  postgres_data:

networks:
  mynetwork:
    external: true
    name: network_lab4